src/Model.cc

```diff
@@ -281,153 +292,191 @@ Errors Model::Load(sdf::ElementPtr _sdf, const ParserConfig &_config)
     }
   }
 
-  // Load all the links.
-  Errors linkLoadErrors = loadUniqueRepeated<Link>(_sdf, "link",
-    this->dataPtr->links, _config);
-  errors.insert(errors.end(), linkLoadErrors.begin(), linkLoadErrors.end());
-
-  // Check links for name collisions and modify and warn if so.
-  for (auto &link : this->dataPtr->links)
+  for (auto elem = _sdf->GetFirstElement(); elem; elem = elem->GetNextElement())
   {
-    std::string linkName = link.Name();
-    if (frameNames.count(linkName) > 0)
+    const std::string &elementName = elem->GetName();
+    if (elementName == "model")
     {
-      // This link has a name collision
-      if (sdfVersion < gz::math::SemanticVersion(1, 7))
+      auto model = loadSingle<Model>(errors, elem, _config);
+      if (!recordUniqueName(nestedModelNames, elementName, model.Name()))
+      {
+        continue;
+      }
+      if (model.IsMerged())
       {
-        // This came from an old file, so try to workaround by renaming link
-        linkName += "_link";
-        int i = 0;
-        while (frameNames.count(linkName) > 0)
+        sdf::Frame proxyFrame = model.PrepareForMerge(errors, "__model__");
+        this->AddFrame(proxyFrame);
+
+        auto moveElements = [](const auto &_src, auto &_dest)
         {
-          linkName = link.Name() + "_link" + std::to_string(i++);
-        }
-        std::stringstream ss;
-        ss << "Link with name [" << link.Name() << "] "
-           << "in model with name [" << this->Name() << "] "
-           << "has a name collision, changing link name to ["
-           << linkName << "].";
-        Error err(ErrorCode::WARNING, ss.str());
-        enforceConfigurablePolicyCondition(
-            _config.WarningsPolicy(), err, errors);
-        link.SetName(linkName);
+          std::move(_src.begin(), _src.end(), std::back_inserter(_dest));
+        };
+        moveElements(model.dataPtr->links, this->dataPtr->links);
+        moveElements(model.dataPtr->frames, this->dataPtr->frames);
+        moveElements(model.dataPtr->joints, this->dataPtr->joints);
+        moveElements(model.dataPtr->models, this->dataPtr->models);
+        moveElements(model.dataPtr->interfaceModels,
+                     this->dataPtr->interfaceModels);
+        moveElements(model.dataPtr->mergedInterfaceModels,
+                     this->dataPtr->mergedInterfaceModels);
```

Comment:
should the names of all these elements be added to `implicitFrameNames`?

Reply:
Done in https://github.com/gazebosim/sdformat/pull/1293/commits/b4c352287333e85b69fa5ce96cc185a5cd9d006b